### ISSUE TITLE
Hide menu bar on fullscreen

### ISSUE THREAD

**Issue by suhr** [wereturtle/ghostwriter#40]
This will make it even less distractive. The margin menu bar adds is useful though.

**Comment by wereturtle** [comment]
I had originally considered this in the design phase for the app, but chose against it.  Generally, user interfaces that disappear and play "peek-a-boo" can confuse new users.  I opted instead to fade the text color of the menu bar so that it is less distracting.

**Comment by suhr** [comment]
> I opted instead to fade the text color of the menu bar so that it is less distracting.

This is still kinda distracting though. Can you just add an option? It can be disabled by default, so no new users would be hurt.

**Comment by wereturtle** [comment]
Well, there are some technical reasons also behind why I didn't do it.  There are two ways I tried to go about it:  One was to use QGraphicsEffect to add a transparency effect to the relevant widgets, and the other is to just outright hide widgets whenever they need to go away.  I already use QGraphicsEffects for the menu bar and the bottom bar for theme effects, and Qt has a limitation where you can't have more than one QGraphicsEffect applied to the same widget.  So that method is out.  If I just outright make the menu bar invisible, I've had some bad experiences with the accelerator keys being nonfunctional.  Then there's the FocusWriter way, which simply removes the widgets and adds them back whenever the mouse drifts to the top of the screen.  It's quite a bit of code, and obviously the accelerator keys still go missing, in which case I have to map it to the main window.  I don't like removing the widgets entirely, as the editor will expand to fill the empty space.

I also wanted to make my application touch screen friendly.  Not everyone has a mouse with which to unhide the widgets.  Sure, you can bind the unhide feature to a shortcut key, but a new user checking off the hide feature won't remember it exists.

Bottom line:  It's just a lot of work to make it happen, and I don't have a lot of spare time.  I'm very happy with the tool where it's at.  It looks great, and doesn't distract me personally.  But anybody can put in a pull request if they want the feature badly enough.  It's something I may try out again in the distant future, but not anytime soon.

**Comment by wereturtle** [comment]
All right.  I have spent quite a few hours tinkering with and analyzing this feature.  In order to do it in a very polished manner (which I would prefer to do, as it is the right way to do anything in life), it's going to take much more work than I have time for.  This is because if I add this feature, I really, really want it to auto-fade the menu bar in and out (animated).  This effect should happen as follows:

* The user stops typing after a few seconds (fade in menu bar - tablet friendly)
* The user resumes typing (fade out menu bar)
* The user hovers the mouse over the menu bar area (fade in)
* The user removes the mouse from the menu bar area (fade out)
* The user activates a keyboard accelerator (fade in)
* The user completes selecting menu option via keyboard (fade out)

I have to rearrange how the background color is applied on top of any background image in order to make this work.  I also need to daisy chain the graphics effects (from QGraphicsEffect).  That means making a QGraphicsEffect derivative that can apply multiple effects, since only one QGraphicsEffect can be applied to a widget at a time.  This means in order to retain the drop shadow effect on the menu text, in addition to adding the fade effect, I would have to implement the drop shadow effect from scratch, rather than using the Qt library.

Sure, I could drop the entire fade animation and just set the text to transparent while removing the drop shadow effect, but that is not nearly as polished.  I see it more as a band-aid solution.  Regardless of how I go about it (band-aid or polished), the above use cases still apply, and will take time to thoroughly test.

The truth is, I already have a day job, and am about to start working a lot of overtime for quite some time.  This feature is probably not going to happen, at least so long as I don't have any additional volunteers to help.  The same goes for the other feature requests you have added.  As such, I am going to close these issues.  Perhaps in the distant future I will have the time.  In the meantime, I am more interested in spending my limited time stomping out the bugs or adding only very simple features.  

Thank you nevertheless for your input.

**Comment by wereturtle** [comment]
Due to another user requesting the feature, I will be implementing it.  However, it will be the "band-aid" way rather than the polished, fade-in/fade-out way.

**Comment by wereturtle** [comment]
Fixed in commit 4e507e0f08207a895fa2e333b431d63a95ebc79d
